### PR TITLE
fix(resource-instance-form): disable "next" button when form is not valid (DEV-803)

### PR DESCRIPTION
**Pull request by mdelez**
resolves DEV-803

### CODE REVIEW

**Comment by mdelez** [comment]
https://user-images.githubusercontent.com/60604010/163142676-940713f6-a6a6-4f92-882a-1ebd9f755d3c.mov

@kilchenmann I also added a progress indicator to clarify that something is happening in the background after the user selects something. If you want to try on your end, I recommend throttling your internet speed in the dev tools.

File: src/app/workspace/resource/resource-instance-form/resource-instance-form.component.html [hunk 1]
```diff
@@ -47,8 +47,10 @@
         </span>
         <span class="fill-remaining-space"></span>
         <span>
+            <!-- check each 'selected' property in addition to the forms validity because the form is actually valid when fetching ontologies and resource classes  -->
             <button mat-raised-button type="button" color="primary"
-                [disabled]="!selectResourceForm.valid || this.errorMessage" (click)="nextStep()" class="form-next">
+                [disabled]="(!selectedProject || !selectedOntology || !selectedResourceClass || !selectResourceForm.valid) || this.errorMessage" (click)="nextStep()" class="form-next">
+                <app-progress-indicator *ngIf="loading" [status]="0" class="next-progress"></app-progress-indicator>
```
Comment (kilchenmann) [inline src/app/workspace/resource/resource-instance-form/resource-instance-form.component.html:53]: Can you set the color of the progress-indicator to primary? Just add `[color]="'primary'" `

**Review by kilchenmann — COMMENTED** [review]
Just a small detail 😉

File: src/app/workspace/resource/resource-instance-form/resource-instance-form.component.html (hunk 1 above)
Comment (mdelez) [reply src/app/workspace/resource/resource-instance-form/resource-instance-form.component.html]: done in [49ddae3](https://github.com/dasch-swiss/dsp-app/pull/713/commits/49ddae3bb591887ff56610b61b40ea6e3ec932f2)

**Review by kilchenmann — APPROVED** [review]

**Pull request merged**